pilot/pkg/model/authentication.go

```diff
@@ -245,3 +247,27 @@ func constructgRPCCallCredentials(tokenFileName, headerKey string) []*core.GrpcS
 		},
 	}
 }
+
+type fbMetadataAnyKey struct {
+	tokenFileName string
+	headerKey     string
+}
+
+var fileBasedMetadataConfigAnyMap = map[fbMetadataAnyKey]*types.Any{}
+
+// findOrMarshalFileBasedMetadataConfig searches google.protobuf.Any in fileBasedMetadataConfigAnyMap
```

Comment:
It would be great to add more comments to explain why we need this function/what problem it solves.

Reply:
Added comment explaining why we need this and what problem it solves. Thanks.